pkg/util/azureclient/storage/storage.go

```diff
@@ -52,6 +52,11 @@ type blobStorageClient struct {
 	storage.BlobStorageClient
 }
 
+// NewBlobStorageClient just for mocking purposes
+func NewBlobStorageClient() BlobStorageClient {
```

Comment:
AFAICS this isn't used in the PR?  I'm a little reluctant about this because the upstream API doesn't have this functionality.

Reply:
we need the mock function NewMockBlobStorageClient(), the mock generator only makes functions based on public functions, so if i remove this i can't create my own mock blob client..